### ISSUE TITLE
Duplicate Servlet Mapping Path /VAADIN/push

### ISSUE THREAD

**Issue by mvysny** [vaadin/flow#19888]
### Description of the bug

Vaadin 23.4.1; spring-boot-skeleton-starter.

When I create my own Spring Servlet such as
```java
@Component
@WebServlet(urlPatterns = "/*", asyncSupported = true, loadOnStartup = 1)
public class MyServlet extends SpringServlet {
    public MyServlet(@Autowired ApplicationContext context) {
        super(context, true);
    }
}
```
push stops working. The following is logged to the console: `Duplicate Servlet Mapping Path /VAADIN/push`

Further analysis: even though I have defined my own servlet in the app, `SpringBootAutoConfiguration.servletRegistrationBean()` still loads the `SpringServlet` anyway and tells it to map PUSH to `/VAADIN/push`. Later on, `JSR356WebsocketInitializer` goes through all Vaadin servlets and tries to initialize Atmosphere for all of those. It discovers both `MyServlet` and `SpringServlet` registered by `SpringBootAutoConfiguration` and tries to initialize atmosphere and map it to `/VAADIN/push`. Second Atmosphere initialization fails: Tomcat throws `DeploymentException` at `JSR356AsyncSupport:105`, which is then logged to the log. Consequently, any attempt to open websocket communication on `/VAADIN/push` fails and can be seen in the browser.

When I create my own Spring Servlet, it conflicts with the Vaadin Spring Servlet loaded by default since both try to initialize Atmosphere to

### Expected behavior

Vaadin should not attempt to register its own servlet if I am trying to register my own.

### Minimal reproducible example

1. git clone skeleton-starter-flow-spring and checkout the v23 branch
2. Add `MyServlet` as described above
3. Run the app

### Versions

- Vaadin / Flow version: 23.4.1
- Java version: 17
- OS version: Ubuntu 22.04
- Browser version (if applicable): Firefox
- Application Server (if applicable): Spring Boot, Tomcat
- IDE (if applicable): IDEA

**Comment by mcollovati** [comment]
I suppose your servlet should "override" the default servlet bean. Try to use `@Component("springServlet")`
An alternative could be defining a `ServletRegistrationBean<SpringServlet>` bean.

**Comment by mcollovati** [comment]
BTW, the double registration has been fixed in 24.3 by adding a `@ConditionalOnMissingBean` on the `SpringServlet` registration (https://github.com/vaadin/flow/pull/17883).
Still, the replacing bean must be named `springServlet` if mapped on `/*`, otherwise the `ServletForwardinController` defined in `VaadinServletConfiguration` will cause `No servlet with name 'springServlet' defined in web.xml` exceptions.

**Comment by TatuLund** [comment]
Atmosphere supports WebSocket only with one servlet. However it should be possible to use LONG_POLLING with multiple servlets.

Related to: https://github.com/vaadin/framework/issues/9013#issuecomment-1822230763

**Comment by mvysny** [comment]
Thank you kindly for your great tips! Please find my responses below:

1. The `@Component("springServlet")` unfortunately doesn't help - or rather it solves the issue by always using the `com.vaadin.flow.spring.SpringServlet`. `MyServlet` is then constructed, but the `serve()` method is never called, rendering the servlet unused.
2. I couldn't get the `ServletRegistrationBean<SpringServlet>` solution to work: Spring always complained that `There is already bean servletRegistrationBean bound`.
3. For some reason `@Push(transport = Transport.LONG_POLLING)` setting was ignored: Vaadin still ran the code in`JSR356WebsocketInitializer`, initialized the websocket and polled `ws://localhost:39117/VAADIN/` in the browser.

Additional notes follow:

I realized that this bug may only occur randomly, since HashMap iteration order in `JSR356WebsocketInitializer` is arbitrary and may change between runs.

The final note is very important. By introducing my own servlet, I now have two servlets running: `MyServlet` listening on `/*` and `SpringServlet` listening on `/vaadinServlet/*`. Both are serving the same set of routes. However, if `MyServlet` introduces some kind of dynamic security rules, then only the app at `/*` will be protected by them - the servlet running at `/vaadinServlet/*` is a vanilla flow SpringServlet which will not have the same security setup, potentially opening a security hole to the app.

**Comment by mvysny** [comment]
> Atmosphere supports WebSocket only with one servlet.
I agree. By introducing my own `MyServlet` I would expect the default Flow SpringServlet to be automatically disabled, exactly as is the case in a non-Spring project with the `VaadinServlet`.

**Comment by mcollovati** [comment]
@mvysny I suppose the workarounds do not work because the `@ConditionalOnMissingBean` fix has not been backported to 23.4

**Comment by mvysny** [comment]
@mcollovati you're right. Would it be possible to backport the `@ConditionalOnMissingBean` fix to 23.x (newest commercial Vaadin works for us too)?

**Comment by mcollovati** [comment]
The change has been back-ported and will be available in the upcoming 23.5.5 release
